app/models/user.rb

```diff
@@ -1,17 +1,31 @@
 class User < ApplicationRecord
-  # Include default devise modules. Others available are:
-  # :confirmable, :lockable, :timeoutable and :omniauthable
+  enum status: [:enabled, :disabled]
+  enum role: [:user, :manager, :admin]
+
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
+         
+  scope :all_users_except_admin, -> { all.where.not(role: :admin) }
 
   has_one :photo, class_name: 'Attachment', as: :attachable
   accepts_nested_attributes_for :photo
 
-  validates :username, presence: true
-  validates :username, length: { in: 6..15 }
+  validates :username, presence: true, length: { in: 6..15 }
```

Comment:
As per verbal discussion, rename it to `name`

Reply:
Sir, Instead of renaming I have logged in using both username and email as suggested.